applications/tari_base_node/src/builder.rs

```diff
@@ -467,12 +467,45 @@ where
 
     // Set the base node for the wallet to the 'local' base node
     let base_node_public_key = base_node_comms.node_identity().public_key().clone();
-    wallet_handles
+    let mut transaction_service_handle = wallet_handles
         .get_handle::<TransactionServiceHandle>()
-        .expect("TransactionService is not registered")
+        .expect("TransactionService is not registered");
+    transaction_service_handle
         .set_base_node_public_key(base_node_public_key.clone())
         .await
         .expect("Problem setting local base node public key for transaction service.");
+    transaction_service_handle
+        .restart_transaction_protocols()
```

Comment:
This is the actual context; it looks like the transaction protocols are started before initial sync has been achieved, unlike `.restart_broadcast_protocols()`

Reply:
Transaction protocols are the negotiation protocols with other wallets so this means that the wallet is immediately available to receive more transactions, any broadcast and chain monitoring protocols that start from scratch before the BN is synced are fine, its the restart of those protocols before the bn is synced that is the problem.

Reply:
Ok point taken, to avoid any potential issues in the BN wallet I have made this change.